### ISSUE TITLE
Log in to Groovestats and retrieve API key within launcher client

### ISSUE THREAD

**Issue by rogerclark** [GrooveStats/gslauncher#7]
The login flow could be simplified by making the login request in the launcher, retrieving the API key token from the request, and saving it locally.

**Comment by Roujo** [comment]
That would imply typing in my password in the launcher, which I'd rather not do. If this is implemented, I'd like to keep the current flow as an option as well. =)

-------- Original Message --------

**Comment by rogerclark** [comment]
Why would you be okay with typing it into the GrooveStats website but not an application vended by GrooveStats?

**Comment by Roujo** [comment]
I can more easily check that I'm on the right website than verify that the binary I'm launching has not been tampered with, and I trust my browser to do the transaction securely more than I trust the launcher. ^^

Mind you, if the password isn't stored beyond the request made to fetch the API key, I can always do that binary verification (with a hash or something) once at that time, so it's not that much of an issue. I can always fill it in myself outside of the launcher, even.

On another note, this makes me realize that the launcher is global to the whole StepMania app while the API key is specific to a certain local profile. To be able to fetch the API key from the launcher, the latter would have to also fetch the local profiles and provide a way for the user to specify which one should use the newly fetched API key - perhaps through a dropdown menu or something.

-------- Original Message --------

**Comment by teejusb** [comment]
I feel like one of the goals of the launcher is to have minimum interaction with it once the game is launched. It won't have knowledge of which profile to attach that API key to as the launcher is player agnostic.

I do have thoughts on how to solve this issue though. E.g. Login methods from straight within the theme (using a QR code to sign in from your phone which can then relay the info back to your machine).

**Comment by archiveflax** [comment]
Yeah, I pitched the QR stuff a while back; other games (Neon FM for example) have done similar things in the past.  The current method was just the simplest to spin up in the timeframe we wanted to hit.  I think we're all interested in simplifying the process in the future.
